src/analyses/uninit.ml

```diff
@@ -135,7 +135,7 @@ struct
     in
     let utar, uoth = unrollType target, unrollType other in
     match ofs, utar, uoth with
-    |     `NoOffset,              _ ,               _ when utar == uoth  -> [v, rev cx]
+    |     `NoOffset,              _ ,               _ when CilType.Typ.equal utar uoth  -> [v, rev cx]
```

Comment:
In general, `CilType.Typ.equal` should only be called if one of the arguments is known to be non-recursive.

Reply:
The modules in `CilType` are precisely defined to be safely called for all arguments. The recursion in `typ` is via `compinfo` which doesn't use derived equality but has one defined just via `ckey` (in `CilType.Compinfo`) to avoid the issue. This was done in #840 a long time ago.

Reply:
I guess I haven't kept up with all the changes in Goblint.